### PR TITLE
Improve FX group doc wording

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Improve FX group doc wording to be easier to understand.

### CODE REVIEW

File: docs/components/fx.md [hunk 1]
```diff
@@ -181,16 +181,14 @@ Note that annotations are also possible with [`fx.Annotate`](https://pkg.go.dev/
...
 For example:
 
-```go
-// server/server.go
-type dependencies struct {
-    fx.In
-    Endpoints []Endpoint `group:"server"`
-}
-```
+Here, are two components adding a `server.Endpoint` type to the `server` group (note the `group` label in the `fx.Out` struct).
```
Comment (estherk15) [inline docs/components/fx.md:191]:
```suggestion
Here, two components add a `server.Endpoint` type to the `server` group (note the `group` label in the `fx.Out` struct).
```

File: docs/components/fx.md [hunk 2]
```diff
@@ -209,9 +207,15 @@ type provides struct {
 }
 ```
 
-Here the `server` component's constructor will get, in `deps.Endpoints`, a slice of all `server.Endpoint` instances provided by other components -- in this case, by `todolist` and `users`.
-Value groups must be named, but typically there is only one group of a given type, so the choice of name isn't critical.
-Using the name of the package defining the type (in this case, `server`) is a good convention.
+Here, a component requests all the types added to the `server` group. This take the form of a slice received at
```
Comment (estherk15) [inline docs/components/fx.md:210]:
```suggestion
Here, a component requests all the types added to the `server` group. This takes the form of a slice received at
```

**Review by estherk15 — COMMENTED** [review]
Left a few grammar notes and a clarification question.

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: docs/components/fx.md [hunk 3]
```diff
@@ -181,16 +181,14 @@ Note that annotations are also possible with [`fx.Annotate`](https://pkg.go.dev/
 
 ### Value Groups
 
-[Value groups](https://pkg.go.dev/go.uber.org/fx#hdr-Value_Groups) allow a requirement to be satisfied with a slice of values from different providers.
+[Value groups](https://pkg.go.dev/go.uber.org/fx#hdr-Value_Groups) make it easier to produce and consume many values of
+the same type. In a nutshell, on one side component can add any types into groups and on the other side other components
```
[the remark this answers is not in the document]
Comment (hush-hush) [reply docs/components/fx.md:185]: I reworded this sentence, WDYT ?

**Review by estherk15 — APPROVED** [review]

**Pull request merged**